web/packages/teleport/src/Discover/Server/EnrollEc2Instance/EnrollEc2Instance.tsx

```diff
@@ -0,0 +1,284 @@
+      );
+
+      // Check for already existing EC2 instances.
+      const checkedEc2Instances: CheckedEc2Instance[] = fetchedEc2Instances.map(
+        ec2 => {
+          const instance =
+            ec2InstancesLookupByInstanceId[ec2.awsMetadata.instanceId];
+          if (instance) {
+            return {
+              ...ec2,
+              ec2InstanceExists: true,
+            };
+          }
+          return ec2;
+        }
+      );
+
+      setFetchEc2InstancesAttempt({ status: 'success' });
+      setTableData({
+        currRegion,
+        nextToken,
+        fetchStatus: nextToken ? '' : 'disabled',
+        items: [...data.items, ...checkedEc2Instances],
+      });
+    } catch (err) {
+      const errMsg = getErrMessage(err);
+      setTableData(data);
+      setFetchEc2InstancesAttempt({ status: 'failed', statusText: errMsg });
+      emitErrorEvent(`ec2 instance fetch error: ${errMsg}`);
+    }
+  }
+
+  async function fetchEc2InstanceConnectEndpoints() {
+    const integrationName = (agentMeta as NodeMeta).integration.name;
+
+    setFetchEc2IceAttempt({ status: 'processing' });
+    try {
+      const { endpoints: fetchedEc2Ices } =
+        await integrationService.fetchAwsEc2InstanceConnectEndpoints(
+          integrationName,
+          {
+            region: selectedInstance.awsMetadata.region,
+            vpcId: selectedInstance.awsMetadata.vpcId,
+          }
+        );
+      setFetchEc2IceAttempt({ status: 'success' });
+      return fetchedEc2Ices;
+    } catch (err) {
+      const errMsg = getErrMessage(err);
+      setFetchEc2InstancesAttempt({ status: 'failed', statusText: errMsg });
+      emitErrorEvent(`ec2 instance connect endpoint fetch error: ${errMsg}`);
+    }
+  }
+
+  function clear() {
+    setFetchEc2InstancesAttempt({ status: '' });
+    setTableData(emptyTableData);
+    setSelectedInstance(null);
+  }
+
+  function handleOnProceed() {
+    fetchEc2InstanceConnectEndpoints().then(ec2Ices => {
+      // If there is already an EICE found and it's create-complete or create-in-progress, we can skip the step where we create the EICE.
+      // Instead, we'll show the dialog to create the node.
+      // If it's create-in-progress, the dialog will wait until it's create-complete and then create the node.
+      // If it's create-complete, the dialog will go straight to creating the node.
+      if (
+        (ec2Ices.length > 0 &&
+          ec2Ices.find(e => e.state === 'create-complete')) ||
+        ec2Ices.find(e => e.state === 'create-in-progress')
+      ) {
+        // If there are multiple EICE's found, we only need the first one that is create-complete, and if there are none that are create-complete,
+        // then the first one that is create-in-progress.
+        const createCompleteEice = ec2Ices.find(
+          e => e.state === 'create-complete'
+        );
+        const createInProgressEice = ec2Ices.find(
+          e => e.state === 'create-in-progress'
+        );
```

Comment:
why do we do the check possibly twice? i'm thinking then, the root `if` check should be just `ec2Ice.length > 0` ?

Reply:
This was likely a leftover from a previous implementation, I've refactored this logic